crypto/rand/rand_lib.c

```diff
@@ -696,6 +696,33 @@ static EVP_RAND_CTX *rand_new_drbg(OSSL_LIB_CTX *libctx, EVP_RAND_CTX *parent,
     return ctx;
 }
 
+#ifdef FIPS_MODULE
+static EVP_RAND_CTX *rand_new_crngt(OSSL_LIB_CTX *libctx, EVP_RAND_CTX *parent)
+{
+    EVP_RAND *rand;
+    EVP_RAND_CTX *ctx;
+
+    rand = EVP_RAND_fetch(libctx, "CRNG-TEST", "fips=no");
```

Comment:
Not really sure what the query should be here? should that be an input if it is possible to come from another provider?

Reply:
The FIPS provider can only fetch from it's internal library context.  It is completely incapable of getting algorithms from anywhere else.  The rationale for the _fips=no_ is as an override for any global property query that might be set.  _-fips_ would also work I think.

This code is only built for compiling for the FIPS provider, so non-FIPS usage need not be considered.